### PR TITLE
Display custom Apps' logos 

### PR DESCRIPTION
**Pull request by lkostrowski**
Displays logo from Manifest (available from 3.14) in apps list, app installation and app details

- [x] Fix paddings / sizes
- [x] QA scenario
- [x] First merge new schema in https://github.com/saleor/saleor-dashboard/pull/3751

To be tested with App with logo:
https://saleor-app-products-feed-git-app-logo-saleorcommerce.vercel.app/api/manifest

or any preview URLs from this PR https://github.com/saleor/apps/pull/552#issuecomment-1587529079
(copy preview URL + add `/api/manifest`

QA:
1. Using app that does not contain `brand.logo.default` field in manifest will not display custom logo - will fallback to generic icon 
2. Using app with `brand.logo.default` will use this logo in following places: app installation page, app list - during installation, app list - after installation, app details

Other changes
1. App installation page icons are changed: new Saleor logo + rectangle icons instead circles
2. App icon size in app details page is bigger

### Screenshots

| | |
|--------|--------|
|Zrzut ekranu 2023-06-13 o 09 58 16 | Zrzut ekranu 2023-06-13 o 09 58 10 |
| Zrzut ekranu 2023-06-13 o 09 57 57 | Zrzut ekranu 2023-06-13 o 09 57 52 |
| Zrzut ekranu 2023-06-13 o 09 57 36 | Zrzut ekranu 2023-06-13 o 09 57 29 |

### Pull Request Checklist

1. [x] This code contains UI changes
4. [x] All visible strings are translated with proper context including data-formatting
5. [ ] Attributes `data-test-id` are added for new elements
6. [ ] The changes are tested in Chrome/Firefox/Safari browsers and in light/dark mode
7. [x] Your code works with the latest stable version of the core

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
3. [ ] giftCard
4. [ ] category
5. [ ] collection
6. [ ] attribute
8. [ ] productType
9. [ ] shipping
10. [ ] customer
11. [ ] permissions
12. [ ] menuNavigation
13. [ ] pages
14. [ ] sales
15. [ ] vouchers
16. [ ] homePage
17. [ ] login
18. [ ] orders
19. [ ] products
20. [x] app
21. [ ] plugins
22. [ ] translations
23. [ ] navigation
24. [ ] variants
25. [ ] payments

CONTAINERS=1

### CODE REVIEW

File: src/apps/components/AppInstallPage/AppInstallPage.tsx [hunk 1]
```diff
@@ -39,6 +41,18 @@ export const AppInstallPage: React.FC<AppInstallPageProps> = ({
 }) => {
   const intl = useIntl();
   const classes = useStyles();
+  const { theme } = useTheme();
+
+  const getSaleorLogoUrl = () => {
```
Comment (krzysztofwolski) [inline src/apps/components/AppInstallPage/AppInstallPage.tsx:46]: We already have a themeable Saleor logo in the side navigation. Is the code duplicated, or was there another reason?

**Review by krzysztofwolski — APPROVED** [review]

File: src/apps/components/AppInstallPage/AppInstallPage.tsx (hunk 1 above)
Comment (lkostrowski) [reply src/apps/components/AppInstallPage/AppInstallPage.tsx:46]: Previously it was using a fixed image. In this diff I replaced it to one that is used in the navigation. Its not themable - these are 2 different PNGs

**Comment by wojteknowacki** [comment]
App img during the installation process is not visible on the app list view from the start - default img is visible and then a new one is placed in its place or sometimes new img is not replacing the default one at all

https://github.com/saleor/saleor-dashboard/assets/124166231/4783bb37-e496-48f9-a880-ea2ae96eea1f

**Comment by wojteknowacki** [comment]
Not all apps placeholders images are visible on the list view after installation - for example slack 
Zrzut ekranu 2023-06-13 o 16 49 31
Not sure is it related to this PR changes or not

**Comment by lkostrowski** [comment]
> There is only one view that does not have an app img - Manage App - is it on purpose? Zrzut ekranu 2023-06-13 o 16 28 45

It was never there, but its a good time add it 👍

**Comment by lkostrowski** [comment]
@wojteknowacki build fill fail - I have to fix tests, but:

1. I used the sam navbar in Manage App page now
2. I fixed not loading Slack and other "official apps" icons. They will fallback to generic defaults. Its on purpose, because once apps are released, source o images will be changed. In this PR it will be regression but on the release it will be invisible for the user
3. Delayed loading of images is known issue. I added separate ticket for it https://github.com/saleor/apps/issues/562. I think its minor. When used refresh pages/navigate back, logos will be loaded. Its caused by Saleor generating logos async

File: src/apps/components/AppDetailsPage/Header.test.tsx
Comment (lkostrowski) [inline src/apps/components/AppDetailsPage/Header.test.tsx:1]: Removed test. It tests nothing and is very flaky

**Review by wojteknowacki — APPROVED** [review]

**Review by krzysztofwolski — DISMISSED** [review]

File: src/fragments/apps.ts [hunk 2]
```diff
@@ -17,6 +17,11 @@ export const appManifestFragment = gql`
       code
       name
     }
+    brand {
+      logo {
+        default
+      }
+    }
```
[the remark this answers is not in the document]
Comment (lkostrowski) [reply src/fragments/apps.ts:24]: TIL :) I will update it

**Review by Droniu — DISMISSED** [review]

**Review by andrzejewsky — APPROVED** [review]

File: src/fragments/apps.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (lkostrowski) [reply src/fragments/apps.ts]: fixed

**Pull request merged**